testsuite/features/support/twopence_init.rb

```diff
@@ -288,3 +281,51 @@ def file_inject(node, local_file, remote_file)
                   'sle12sp4_terminal'         => $sle12sp4_terminal,
                   'sle15sp3_buildhost'        => $sle15sp3_buildhost,
                   'sle15sp3_terminal'         => $sle15sp3_terminal }
+
+# This is the inverse of `node_by_host`.
+# For each node we choose as key the most meaninful host, i.e. we drop aliases.
+# For instance, we ignore the alias 'ssh_spack_migrated_minion`, we use 'ssh_minion`.
+$host_by_node = {}
+$node_by_host.each do |host, node|
+  next if node.nil?
+  next if node.is_a?(String) && node.empty?
+
+  ignored_hosts = %w[sle_migrated_minion sle_ssh_tunnel_client
+                     sle_spack_migrated_minion sle_ssh_tunnel_minion
+                     ceos_ssh_minion ubuntu_ssh_minion]
+  next if ignored_hosts.include? host
+
+  $host_by_node[node] = host
+  puts "Added #{host} as key for node: '#{node}'"
+end
+
+def client_public_ip(host)
+  node = $node_by_host[host]
+  interface = case host
+              when /^sle/, /^ceos/, /^debian/, 'server', 'proxy', 'build_host'
+                'eth0'
+              when /^ubuntu/
+                'ens3'
+              else
+                raise "Unknown net interface for #{host}"
+              end
+  output, code = node.run("ip address show dev #{interface} | grep 'inet '")
+  raise 'Cannot resolve public ip' unless code.zero?
+
+  output.split[1].split('/')[0]
+end
+
+# Initialize IP address or domain name
+$nodes.each do |node|
+  next if node.nil?
+  next if node.is_a?(String) && node.empty?
+
+  node.init_ip(node.full_hostname)
```

Comment:
Not for this PR, but I think we need a card to research and address if this instantiation is ok, where we use .ip attribute, etc.

Reply:
I was forced to add this guard after realizing that some nodes were passing with value the empty string.
The code creating those nodes should be responsible to change those empty strings to `nil`.

Reply:
OK, but I'm not talking about guard :)
I'm talking about setting inside the .ip attribute the string .full_hostname